neuralmonkey/writers/plain_text_writer.py

```diff
@@ -1,34 +1,58 @@
 from typing import Iterator, List, Any, Callable
 
 from neuralmonkey.logging import log
+from neuralmonkey.readers.plain_text_reader import _ALNUM_CHARSET
 
 # pylint: disable=invalid-name
 # Writer: function that gets file and the data
 Writer = Callable[[str, Any], None]
 # pylint: enable=invalid-name
 
 
-def tokenized_text_writer(encoding: str = "utf-8") -> Writer:
+def text_writer(encoding: str = "utf-8") -> Writer:
 
-    def writer(path: str, data: Iterator[List[str]]) -> None:
+    def writer(path: str, data: Iterator[Any]) -> None:
         with open(path, "w", encoding=encoding) as f_out:
             for sentence in data:
-                f_out.write(" ".join(sentence) + "\n")
-        log("Result saved as tokenized text in '{}'".format(path))
+                f_out.write(str(sentence) + "\n")
+        log("Result saved as plain text in '{}'".format(path))
 
     return writer
 
 
-def text_writer(encoding: str = "utf-8") -> Writer:
+def tokenized_text_writer(encoding: str = "utf-8") -> Writer:
 
-    def writer(path: str, data: Iterator[Any]) -> None:
-        with open(path, "w", encoding=encoding) as f_out:
-            for sentence in data:
-                f_out.write(str(sentence) + "\n")
-        log("Result saved as plain text in '{}'".format(path))
+    def writer(path: str, data: Iterator[List[str]]) -> None:
+        writer = text_writer(encoding)
+        writer(path, (" ".join(s) for s in data))
+
+    return writer
+
+
+def t2t_tokenized_text_writer(encoding: str = "utf-8") -> Writer:
+    """Get a writer that is reversed to the readers.t2t_tokenized_text_reader.
+
+    Code in the detokenize method is inspired by tensor2tensor tokenizer:
+    https://github.com/tensorflow/tensor2tensor/blob/v1.5.5/tensor2tensor/data_generators/text_encoder.py
+    """
+    def detokenize(data: Iterator[List[str]]) -> Iterator[str]:
+        for sentence in data:
+            is_alnum = [t[0] in _ALNUM_CHARSET for t in sentence]
```

Comment:
tadyto `is_alnum` se nastaví jen na prvním znaku z celý věty a pak už se nenastaví. Tohle měl odhalit unit test -> write a unit test for this function pls

Reply:
Pro kazdy token chces vedet, jestli se sklada jen z is_alnum znaku nebo (! is_alnum) znaku. Pokud tohle pouzivas v kombinaci s t2t_readerem, ktery provadi presne tuto tokenizaci + t2t vocabulary, ktery obsahuje jen takove tokeny, tak ti staci kouknout na prvni znak tokenu

Reply:
jo sorry nevšim sem si že je to list comprehension.